plugin/build.gradle.kts

```diff
@@ -140,6 +148,11 @@ dependencies {
     }
 }
 
+// This is necessary to add android build tools to the test runtime classpath
+tasks.withType<PluginUnderTestMetadata>().configureEach {
+    pluginClasspath.from(configurations["testDependencies"])
+}
```

Comment:
This is typically handled by having a `compileOnly` and `testRuntimeOnly` from a testing standpoint.

Reply:
We use gradle's TestKit for testing plugins which doesn't use `testRuntimeOnly` when building the classpath the plugin is tested with. With the above code, you'd see something like
```
* Exception is:
java.lang.NoClassDefFoundError: com/android/build/api/dsl/CompileOptions
	at org.openrewrite.gradle.DelegatingProjectParser.<init>(DelegatingProjectParser.java:65)
	at org.openrewrite.gradle.AbstractRewriteTask.getProjectParser(AbstractRewriteTask.java:89)
```

The issue and solution is described here: https://discuss.gradle.org/t/issue-in-a-custom-plugin-with-plugins-dsl-and-android-plugin-dependency/29123

Reply:
Ahh, right, right. I've also used the method to add additional `testSourceSets` to handle that issue as well.

Reply:
The need for this workaround was removed with the current approach of adding the plugin class loader to `RewriteClassLoader`